### PR TITLE
C1 see contacts2

### PR DESCRIPTION
**Pull request by DanielGardiner**
We have changed the way that the hiring partner details are displayed to being in a modal. We have also added a list of hiring partner contacts to this.
We updated the hiring partner model to get contacts by company and get details by company.
We updated displayHiringPartnerPageController.
Made a hiring partner details modal controller and factory.
We updated hiring partner page view to include a modal which we made in templates and a JS file to enact it. We also changed the JS for the hiring partner page js.
Added unit test

### CODE REVIEW

File: public/js/hiringPartner.js [hunk 1]
```diff
@@ -145,47 +145,28 @@ async function getHiringPartners () {
...
 /**
  * Runs a foreach through each hiring partner object and outputs HTML elements with hiring partner's details
  *
  * @param partnerCompanies is an array of objects which contains information about hiring partners
  */
 function displayHiringPartnerHandler(partnerCompanies){
-    let companyDisplayer = document.getElementById('companies')
-    let companyInformation = ''
+    let companyNamesHTML = '';
+    console.log('hello')
```
Comment (ashleycoles) [inline public/js/hiringPartner.js]: I'm not sure we should be console logging hello on master

File: public/js/hiringPartnerModal.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+                let yesNoSub = ['No', 'Yes']
+
+                data.forEach(function (contact) {
+
+                    $validatedName = validateContactField(contact.name)
+                    $validatedEmail = validateContactField(contact.email)
+                    $validatedJobTitle = validateContactField(contact.job_title)
+                    $validatedPhone = validateContactField(contact.phone)
+                    $validatedIsPrimaryContact = validateContactField(yesNoSub[contact.is_primary_contact])
```
Comment (ashleycoles) [inline public/js/hiringPartnerModal.js]: Variables don't need a dollar in JS

File: src/Classes/Models/HiringPartnerModel.php [hunk 3]
```diff
@@ -175,4 +176,19 @@ public function getHiringPartners() :array
...
+     * Gets a single hiring partners information
+     *
+     * @return array array with the info
+     */
+    public function getDetailsByCompany($id) :array
+    {
+        $query = $this->db->prepare("SELECT 
+							  	`id`, `name`, `size`, `tech_stack`, `postcode`, `phone_number`, `url_website`
+								FROM `hiring_partner_companies`
+								WHERE `id` = :id;");
```
Comment (ashleycoles) [inline src/Classes/Models/HiringPartnerModel.php]: Check whitespace

File: src/Classes/Models/HiringPartnerModel.php [hunk 4]
```diff
@@ -175,4 +176,19 @@ public function getHiringPartners() :array
         $query->execute();
         return $query->fetchAll();
     }
+
+    /**
+     * Gets a single hiring partners information
+     *
+     * @return array array with the info
+     */
+    public function getDetailsByCompany($id) :array
```
Comment (ashleycoles) [inline src/Classes/Models/HiringPartnerModel.php]: Type hint param

File: src/settings.php [hunk 5]
```diff
@@ -17,7 +17,7 @@
         ],
 
         'db' => [
-            'host' => 'mysql:host=db;',
+            'host' => 'mysql:host=127.0.0.1;',
```
Comment (ashleycoles) [inline src/settings.php]: Please change the DB creds back to the original db

File: public/js/hiringPartner.js (hunk 1 above)
Comment (DanielGardiner) [reply public/js/hiringPartner.js]: Done

File: public/js/hiringPartnerModal.js (hunk 2 above)
Comment (DanielGardiner) [reply public/js/hiringPartnerModal.js]: done

File: src/Classes/Models/HiringPartnerModel.php (hunk 3 above)
Comment (DanielGardiner) [reply src/Classes/Models/HiringPartnerModel.php]: done

File: src/Classes/Models/HiringPartnerModel.php (hunk 4 above)
Comment (DanielGardiner) [reply src/Classes/Models/HiringPartnerModel.php]: done

File: src/settings.php (hunk 5 above)
Comment (DanielGardiner) [reply src/settings.php]: done

**Pull request merged**